android-plugin/src/main/java/com/zendesk/unity/ZDK_Plugin.java

```diff
@@ -44,10 +44,15 @@
     private static final int CONTACT_US_BUTTON_VISIBILITY_OFF = 0;
     private static final int CONTACT_US_BUTTON_VISIBILITY_ARTICLE_LIST_ONLY = 1;
     private static final int CONTACT_US_BUTTON_VISIBILITY_ARTICLE_LIST_ARTICLE_VIEW = 2;
+
+    // Original behavior was to have article voting enabled
+    private boolean _articleVotingEnabled = true;
     
     public static ZDK_Plugin _instance;
     public static Object instance(){
-        _instance = new ZDK_Plugin();
+        if (_instance == null) {
```

Comment:
Not having a singleton approach here was causing state to get lost.
I could have made my variables static but I prefer to handle a singleton pattern here following that the `instance` is basically self documenting it as such.

Reply:
Hey @juank-pa can you give us more detail about what state was being lost?

Reply:
In the original implementation there was no state. But after my changes to make `setArticleVotingEnabled` work on both platforms (borrowed from iOS) I needed to store that state in the instance, which I think is more portable than doing it statically. For that reason keeping the singleton alive became necessary.